lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' as ui;
+
+/// Tracks the [FlutterView]s focus changes.
+final class ViewFocusBinding {
+  ViewFocusBinding._();
+
+  static ViewFocusBinding? _instance;
+
+  /// Initializes the [ViewFocusBinding] singleton.
+  static void initialize() {
+    _instance ??= ViewFocusBinding._().._init();
+  }
+
+  final EnginePlatformDispatcher _dispatcher =
+      EnginePlatformDispatcher.instance;
+
+  void _init() {
+    domDocument.body?.addEventListener(_focusin, _focusChangeHandler, true);
+    domDocument.body?.addEventListener(_focusout, _focusChangeHandler, true);
+    registerHotRestartListener(_dispose);
```

Comment:
You can remove this self-de-registration by initializing/disposing this in the platform_dispatcher:

* init here: https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/platform_dispatcher.dart#L75-L86
* dispose here: https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/platform_dispatcher.dart#L112-L118

I mention this because testing these hot restart listeners is a PITA, and this way you have fewer moving parts.

Reply:
Thank you for the suggestion. Kudos for code pointers :handshake: